Wireshark plugin segfaults Wireshark test suite

This plugin causes two unexpected test failures in Wireshark's test suite, due to a pointer which can be used uninitialized causing a segmentation fault. Patch incoming.

To reproduce:

1. Manually follow the steps of `wireshark/build-scripts/linux_build.sh 4.2.0`, but *removing* the `-DBUILD_wireshark=off` from the CMake command line, to build Wireshark 4.2.0 with the ja4 plugin
2. Notice this warning during the build process: 
```
/.../wireshark-4.2.0/plugins/epan/ja4/packet-ja4.c: In function ‘dissect_ja4’:
/.../wireshark-4.2.0/plugins/epan/ja4/packet-ja4.c:1113:86: warning: ‘packet_time’ may be used uninitialized in this function [-Wmaybe-uninitialized]
```
3. From the `wireshark-4.2.0/build` directory, run `cmake --build . --target test-programs`
4. From the same directory, run `python3 -m pytest . -k 'not sharkd'`
5. Observe the failed test cases (my results are here: [test-output-4.2.0.txt](https://github.com/FoxIO-LLC/ja4/files/15262817/test-output-4.2.0.txt))
6. remove `run/plugins/4.2/epan/ja4.so` and repeat step 4, to see that the removal of the JA4 plugin causes the test suite to pass

(Step 4 needs `-k 'not sharkd'` because some sharkd-related test appears to hard-code a list of expected plugins, and fails when ja4 appears in the result.)

--- a/wireshark/source/packet-ja4.c
+++ b/wireshark/source/packet-ja4.c
@@ -767,7 +767,7 @@ dissect_ja4(tvbuff_t *tvb, packet_info *pinfo _U_, proto_tree *tree, void *dummy
 	nstime_t latency;
 	nstime_t latency2;
 
-	nstime_t *packet_time;
+	nstime_t *packet_time = NULL;
 	int seq = 0;
 	int ack = 0;
 	int syn = 0;
@@ -984,7 +984,7 @@ dissect_ja4(tvbuff_t *tvb, packet_info *pinfo _U_, proto_tree *tree, void *dummy
 				if (fvalue_get_uinteger(get_value_ptr(field)) == 0x02) {
 					syn = 1;
 					conn->client_ttl = curr_ttl;
-					if (conn->timestamp_A.secs == 0) {
+					if ((packet_time != NULL) && (conn->timestamp_A.secs == 0)) {
 						conn->timestamp_A.secs = packet_time->secs;
 						conn->timestamp_A.nsecs = packet_time->nsecs;
 					}
@@ -994,18 +994,18 @@ dissect_ja4(tvbuff_t *tvb, packet_info *pinfo _U_, proto_tree *tree, void *dummy
 				if (fvalue_get_uinteger(get_value_ptr(field)) == 0x012) {
 					syn = 2;
 					conn->server_ttl = curr_ttl;
-					if (conn->timestamp_B.secs == 0) {
+					if ((packet_time != NULL) && (conn->timestamp_B.secs == 0)) {
 						conn->timestamp_B.secs = packet_time->secs;
 						conn->timestamp_B.nsecs = packet_time->nsecs;
 					}
-					if (conn->syn_ack_count <= 10) {
+					if ((packet_time != NULL) && (conn->syn_ack_count <= 10)) {
 						conn->syn_ack_times[conn->syn_ack_count].secs = packet_time->secs;
 						conn->syn_ack_times[conn->syn_ack_count++].nsecs = packet_time->nsecs;
 					}
 				}
 
 				// Add RST for JA4T
-				if (fvalue_get_uinteger(get_value_ptr(field)) == 0x004) {
+				if ((packet_time != NULL) && (fvalue_get_uinteger(get_value_ptr(field)) == 0x004)) {
 					syn = 3;
 					conn->rst_time.secs = packet_time->secs;
 					conn->rst_time.nsecs = packet_time->nsecs;
@@ -1021,7 +1021,7 @@ dissect_ja4(tvbuff_t *tvb, packet_info *pinfo _U_, proto_tree *tree, void *dummy
 					}
 					srcport = dstport = 0;
 
-					if ((conn->timestamp_C.secs == 0) && (seq == 1) && (ack == 1)) {
+					if ((packet_time != NULL) && (conn->timestamp_C.secs == 0) && (seq == 1) && (ack == 1)) {
 						conn->timestamp_C.secs = packet_time->secs;
 						conn->timestamp_C.nsecs = packet_time->nsecs;
 					}
@@ -1032,20 +1032,20 @@ dissect_ja4(tvbuff_t *tvb, packet_info *pinfo _U_, proto_tree *tree, void *dummy
 				// we identify them with PSH, ACK and the direction
 				if (fvalue_get_uinteger(get_value_ptr(field)) == 0x018) {
 					if (conn->server_ttl && conn->client_ttl) {
-						if (conn->timestamp_D.nsecs == 0) {
+						if ((packet_time != NULL) && conn->timestamp_D.nsecs == 0) {
 							// Denotes first PSH, ACK
 							conn->timestamp_D.secs = packet_time->secs;
 							conn->timestamp_D.nsecs = packet_time->nsecs;
 						} else {
 
 
-							if ((srcport < 5000) && (conn->timestamp_E.nsecs == 0)) {
+							if ((packet_time != NULL) && (srcport < 5000) && (conn->timestamp_E.nsecs == 0)) {
 								// Denotes second PSH, ACK - JA4L-S goes here
 								conn->timestamp_E.secs = packet_time->secs;
 								conn->timestamp_E.nsecs = packet_time->nsecs;
 							}
 
-							if ((dstport < 5000) && (conn->timestamp_F.nsecs == 0)) {
+							if ((packet_time != NULL) && (dstport < 5000) && (conn->timestamp_F.nsecs == 0)) {
 								// Denotes third PSH, ACK - JA4L-C goes here
 								conn->timestamp_F.secs = packet_time->secs;
 								conn->timestamp_F.nsecs = packet_time->nsecs;
@@ -1089,12 +1089,12 @@ dissect_ja4(tvbuff_t *tvb, packet_info *pinfo _U_, proto_tree *tree, void *dummy
 
 				// QUIC Initial packets
 				if (fvalue_get_uinteger(get_value_ptr(field)) == 0) {
-					if ((dstport == 443) && (conn->timestamp_A.nsecs == 0)){
+					if ((packet_time != NULL) && (dstport == 443) && (conn->timestamp_A.nsecs == 0)){
 						conn->client_ttl = curr_ttl;
 						conn->timestamp_A.secs = packet_time->secs;
 						conn->timestamp_A.nsecs = packet_time->nsecs;
 					}
-					if ((srcport == 443) && (conn->timestamp_B.nsecs == 0)) {
+					if ((packet_time != NULL) && (srcport == 443) && (conn->timestamp_B.nsecs == 0)) {
 						conn->server_ttl = curr_ttl;
 						conn->timestamp_B.secs = packet_time->secs;
 						conn->timestamp_B.nsecs = packet_time->nsecs;
@@ -1103,12 +1103,12 @@ dissect_ja4(tvbuff_t *tvb, packet_info *pinfo _U_, proto_tree *tree, void *dummy
 
 				// QUIC handshake packets, keep updating C until D is found
 				if (fvalue_get_uinteger(get_value_ptr(field)) == 2) {
-					if ((srcport == 443) && (conn->timestamp_D.nsecs == 0)) {
+					if ((packet_time != NULL) && (srcport == 443) && (conn->timestamp_D.nsecs == 0)) {
 						conn->timestamp_C.secs = packet_time->secs;
 						conn->timestamp_C.nsecs = packet_time->nsecs;
 					}
 
-					if ((dstport == 443) && (conn->timestamp_D.nsecs == 0)){
+					if ((packet_time != NULL) && (dstport == 443) && (conn->timestamp_D.nsecs == 0)){
 						conn->timestamp_D.secs = packet_time->secs;
 						conn->timestamp_D.nsecs = packet_time->nsecs;
 

